controllers/useraccount/useraccount_controller.go

```diff
@@ -302,23 +322,20 @@ func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, userAcc *toolchainv
 				return nil, false, err
 			}
 			nsTmplSet = newNSTemplateSet(userAcc)
-			if err := controllerutil.SetControllerReference(userAcc, nsTmplSet, r.Scheme); err != nil {
-				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err,
-					"failed to set controller reference for NSTemplateSet '%s'", name)
-			}
```

Comment:
Shouldn't we remove the owner reference from the already existing resources?

Reply:
good point. I've added a `migrateNSTemplateSetIfNecessary` in 86145fa to clearly identify this step when processing the UserAccounts. Let me know what you think about it.